### PR TITLE
[21.09] fix value_in_data_table validator for multiple selects

### LINKED ISSUE #12928: value_in_data_table / value_not_in_data_table validator doesn't work with select type parameters
**Describe the bug**
The `value_in_data_table` parameter offers to validate either text or select type parameters from the contents of a tool data table. It however, fails to work as advertised for parameters of type select. The failure modes observed are (1) allowing all inputs (for the case of `multiple="true"` and `value_not_in_data_table` and (2) "flickering" the Execute button but not displaying the relevant message about a validation problem.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version 22.01 dev0
Commit: 88a69285e840fc1e5865d3597e5a1eb709144a9b

**To Reproduce**
1. Run the tool below (in Additional context) with associated data table specification.
2. Try various combinations of `value_not_in_data_table` and `value_in_data_table`.
3. Observe failure

**Expected behavior**
The text input parameter shows an example of what is expected - allow valid values, block and warn on invalid.

**Screenshots**

**Additional context**

```xml
<tool id="test1" name="test_validate" version="0.1">
    <command><![CDATA[
        echo '$input1' > '$output1'
    ]]></command>
    <inputs>
        <param name="seen" type="select" label="Contents">
            <options from_data_table="primer_scheme_bedfiles">
                <column name="value" index="0" />
            </options>
        </param>
        <param name="input1" type="select" label="Input">
            <option value="SARS-CoV-2-ARTICv1" selected="true">SARS-CoV-2 ARTIC v1</option>
            <option value="SARS-CoV-2-ARTICv2" selected="true">SARS-CoV-2 ARTIC v2</option>
            <option value="SARS-CoV-2-ARTICv3" selected="true">SARS-CoV-2 ARTIC v3</option>
            <option value="SARS-CoV-2-ARTICv4" selected="true">SARS-CoV-2 ARTIC v4</option>
            <option value="VarSkip-V1a" selected="true">NEB VarSkip-V1a</option>
            <option value="Midnight-v1" selected="true">RAPID/Midnight v1</option>
            <validator type="value_in_data_table" metadata_column="0" message="MESSAGE" table_name="primer_scheme_bedfiles" />
        </param>

        <param name="input2" type="text" label="Input2">
            <validator type="value_not_in_data_table" metadata_column="0" message="MESSAGE" table_name="primer_scheme_bedfiles" />
        </param>
    </inputs>
    <outputs>
        <data format="txt" name='output1' />
    </outputs>
</tool>
```

```xml
<tables>
    <!-- Locations of primer scheme BED files -->
    <table name="primer_scheme_bedfiles" comment_char="#" allow_duplicate_entries="False">
        <columns>value, description, path</columns>
        <file path="tool-data/primer_scheme_bedfiles.loc" />
    </table>
</tables>
```

### PR DESCRIPTION
**Pull request by bernt-matthias**
The value_in_data_table failed for selects with 'multiple="true"'.

fixes https://github.com/galaxyproject/galaxy/issues/12928

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/tools/parameters/basic.py [hunk 1]
```diff
@@ -1047,6 +1047,15 @@ def to_dict(self, trans, other_values=None):
...
 
+    def validate(self, value, trans=None):
+        if value:
+            if self.multiple:
+                if not isinstance(value, list):
+                    value = [value]
+            else:
+                value = [value]
+            for v in value:
+                super().validate(v, trans)
```
Comment (mvdbeek) [inline lib/galaxy/tools/parameters/basic.py]:
```suggestion
        if not value:
            super().validate(value, trans)
        if self.multiple:
            if not isinstance(value, list):
                value = [value]
        else:
            value = [value]
        for v in value:
            super().validate(v, trans)

```
That should fix the no options validator (and the linting)

**Comment by mvdbeek** [comment]
I'm OK to apply this to 21.09 since it will avoid very unfortunate tool development workarounds

**Comment by bernt-matthias** [comment]
OK. I rebased this to 21.09

**Comment by mvdbeek** [comment]
Thanks a lot @bernt-matthias!

**Pull request merged**